server/src/main/java/io/druid/server/DruidNode.java

```diff
@@ -83,11 +83,10 @@ private void init(String serviceName, String host, Integer port)
     Preconditions.checkNotNull(serviceName);
     this.serviceName = serviceName;
 
-    if(host == null && port == null) {
+    if (host == null && (port == null || port < 0)) {
```

Comment:
One way to handle not setting a port in unit tests is as follows:

```
new Module() {
          @Override
          public void configure(Binder binder)
          {
            JsonConfigProvider.bindInstance(
                binder, Key.get(DruidNode.class, Self.class), new DruidNode("test-inject", null, null)
            );
          }
        }
```

Can a module be added when it is needed as per above instead of changing the logic here?